File: src/leetcode/rotatearray/java/RotateArray.java
package leetcode.rotatearray.java;

/*
Given an integer array nums, rotate the array to the right by k steps, where k is non-negative.
 */

import java.util.Arrays;

public class RotateArray {
    public void rotate(int[] nums, int k) {
        System.out.println("Begin " + Arrays.toString(nums));
        for (int i = 0; i < k; i++) {
            int[] arr = nums.clone();
            for (int j = 0; j < nums.length; j++) {
                if (j == nums.length - 1){
                    nums[0] = arr[j];
                }else {
                    nums[j+1] = arr[j];
                }
            }
        }
        System.out.println("End " + Arrays.toString(nums));
    }
}

File: src/leetcode/rotatearray/kotlin/RotateArray.kt
package leetcode.rotatearray.kotlin

/*
Given an integer array nums, rotate the array to the right by k steps, where k is non-negative.
 */

class RotateArray {
    fun rotate(nums: IntArray, k: Int) {
        println("Begin ${nums.contentToString()}")
        for ( n in 0 until k){
            val aux = nums.clone()
            for (j in nums.indices){
                if (j == nums.size - 1) nums[0] = aux[j]
                else nums[j+1] = aux[j]
            }
        }
        println("End ${nums.contentToString()}")
    }
}

